Subject: Linux-RT 2.6.25-rt
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
commit 4cf900b11bc1e75b0258f4bd3db921316da640d2
Author: Steven Rostedt <rostedt@goodmis.org>
Date:   Sat Feb 9 10:28:09 2008 -0500

    ftrace: don't use raw_local_irq_save/restore
    
    Using raw_local_irq_save/restore confuses lockdep.
    It's fine to use the normal ones.
    
    Signed-off-by: Steven Rostedt <srostedt@redhat.com>

---
 kernel/trace/trace.c              |    4 ++--
 kernel/trace/trace_sched_switch.c |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

Index: linux-2.6.24.4-rt4/kernel/trace/trace.c
===================================================================
--- linux-2.6.24.4-rt4.orig/kernel/trace/trace.c	2008-03-24 19:05:46.000000000 -0400
+++ linux-2.6.24.4-rt4/kernel/trace/trace.c	2008-03-24 19:05:47.000000000 -0400
@@ -269,7 +269,7 @@ function_trace_call(unsigned long ip, un
 	if (unlikely(!tracer_enabled))
 		return;
 
-	raw_local_irq_save(flags);
+	local_irq_save(flags);
 	cpu = raw_smp_processor_id();
 	data = tr->data[cpu];
 	disabled = atomic_inc_return(&data->disabled);
@@ -278,7 +278,7 @@ function_trace_call(unsigned long ip, un
 		ftrace(tr, data, ip, parent_ip, flags);
 
 	atomic_dec(&data->disabled);
-	raw_local_irq_restore(flags);
+	local_irq_restore(flags);
 }
 
 static struct ftrace_ops trace_ops __read_mostly =
Index: linux-2.6.24.4-rt4/kernel/trace/trace_sched_switch.c
===================================================================
--- linux-2.6.24.4-rt4.orig/kernel/trace/trace_sched_switch.c	2008-03-24 19:05:43.000000000 -0400
+++ linux-2.6.24.4-rt4/kernel/trace/trace_sched_switch.c	2008-03-24 19:05:47.000000000 -0400
@@ -30,7 +30,7 @@ ctx_switch_func(struct task_struct *prev
 	if (!tracer_enabled)
 		return;
 
-	raw_local_irq_save(flags);
+	local_irq_save(flags);
 	cpu = raw_smp_processor_id();
 	data = tr->data[cpu];
 	disabled = atomic_inc_return(&data->disabled);
@@ -39,7 +39,7 @@ ctx_switch_func(struct task_struct *prev
 		tracing_sched_switch_trace(tr, data, prev, next, flags);
 
 	atomic_dec(&data->disabled);
-	raw_local_irq_restore(flags);
+	local_irq_restore(flags);
 }
 
 void ftrace_ctx_switch(struct task_struct *prev, struct task_struct *next)
